### ISSUE TITLE
wat project: build error causes app stuck at "building"

### ISSUE THREAD

**Issue by kenrick95** [wasdk/WebAssemblyStudio#174]
1. https://webassembly.studio/?f=qwejfuumnhd
2. Notice that there is error in wat codes (`$add` is not defined); Note: this is intentional
3. Click "Build"
4. App will stuck at "Building project..." and no other action could be made

<img width="724" alt="screen shot 2018-03-22 at 12 03 01 pm" src="https://user-images.githubusercontent.com/3090380/37750636-0ca55306-2dc9-11e8-89e7-8712cbdb0c23.png">

The error at browser console is:
```
Error: resolveNames failed:
test.wat:6:23: error: undefined function variable "$add"
  (export "add" (func $add))
                      ^^^^
```

---

Update: The issue here is that if the build process failed (in this case due to undefined function variable), the WebAssemblyStudio app is stuck at "building"

**Comment by mbebenita** [comment]
This is really strange. If i create an empty .wat project, the main.wat file looks different. The name of the main function is $add, not $error.

**Comment by kenrick95** [comment]
Yeah, that's my point, if I mistakenly have an error in my .wat project, in this case naming the function as `$error` instead of `$add`, seems like the whole WebAssemblyStudio app will crash (and I need to reload the page to recover its functionality, losing the codes that I've worked on)

**Comment by yurydelendik** [comment]
Fixed by #176
